cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -44,11 +44,25 @@ import (
 	utilflag "k8s.io/apiserver/pkg/util/flag"
 )
 
+// deprecatedOption can be used to define a deprecated command line
+// flag. Whenever it is set, regardless to which value, a warning is
+// triggered.
+type deprecatedOption struct {
+	option string
+}
+
+func (d deprecatedOption) String() string { return "" }
+func (d deprecatedOption) Set(value string) error {
+	klog.Warningf("Warning: option %s=%q is deprecated and has no effect", d.option, value)
+	return nil
+}
+func (d deprecatedOption) Type() string { return "" }
```

Comment:
This comment supersedes all my other comments.

in general I'm not sure why we're using a struct here when maybe just a function will do. I think I would prefer that the definition of these deprecated flags more closely mirrored that of a normal flag. Maybe something as simple that would be invoked like this:
`connectionTimeout    = deprecatedflag.Set("connection-timeout")`

Reply:
Yes, the API can be made nicer. Let me try something along those lines.

Reply:
@davidz627 please have another look. I moved the code into an utility package with a single function as interface. Actually, it's two now, because I also added support for boolean flags.